SqlUpdateTest failure

_master_ (commit a26822d3ffd6ec5357d49f0fbeb2cc53eea469ba)

Failed on PR builder: http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/6384/testReport/junit/com.hazelcast.jet.sql.impl.connector.map/SqlUpdateTest/when_updateUnknownMapping_then_fails/

Stacktrace:
```
Expecting throwable message:
  <"Cannot resolve IMap schema because it doesn't have entries on the local member: test_map">
to contain:
  <"Object 'test_map' not found">
but did not.

Throwable that failed the check:

com.hazelcast.sql.HazelcastSqlException: Cannot resolve IMap schema because it doesn't have entries on the local member: test_map
 at com.hazelcast.sql.impl.QueryUtils.toPublicException(QueryUtils.java:79)
 at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:238)
 at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:200)
 at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:196)
 at com.hazelcast.sql.SqlService.execute(SqlService.java:146)
 at com.hazelcast.jet.sql.impl.connector.map.SqlUpdateTest.execute(SqlUpdateTest.java:306)
 at com.hazelcast.jet.sql.impl.connector.map.SqlUpdateTest.lambda$when_updateUnknownMapping_then_fails$6(SqlUpdateTest.java:297)
 at org.assertj.core.api.ThrowableAssert.catchThrowable(ThrowableAssert.java:62)
 at org.assertj.core.api.AssertionsForClassTypes.catchThrowable(AssertionsForClassTypes.java:750)
 at org.assertj.core.api.Assertions.catchThrowable(Assertions.java:1179)
 at org.assertj.core.api.Assertions.assertThatThrownBy(Assertions.java:1051)
 at com.hazelcast.jet.sql.impl.connector.map.SqlUpdateTest.when_updateUnknownMapping_then_fails(SqlUpdateTest.java:297)
 at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
 at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 at java.lang.reflect.Method.invoke(Method.java:498)
 at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
 at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
 at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
 at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
 at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
 at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
 at java.util.concurrent.FutureTask.run(FutureTask.java:266)
 at java.lang.Thread.run(Thread.java:748)
Caused by: com.hazelcast.sql.impl.QueryException: Cannot resolve IMap schema because it doesn't have entries on the local member: test_map
 at com.hazelcast.sql.impl.QueryException.error(QueryException.java:60)
 at com.hazelcast.sql.impl.calcite.parse.QueryParser.parse(QueryParser.java:95)
 at com.hazelcast.sql.impl.calcite.OptimizerContext.parse(OptimizerContext.java:140)
 at com.hazelcast.sql.impl.calcite.CalciteSqlOptimizer.prepare(CalciteSqlOptimizer.java:127)
 at com.hazelcast.sql.impl.SqlServiceImpl.prepare(SqlServiceImpl.java:292)
 at com.hazelcast.sql.impl.SqlServiceImpl.query0(SqlServiceImpl.java:273)
 at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:225)
 ... 22 more
Caused by: com.hazelcast.sql.impl.QueryException: Cannot resolve IMap schema because it doesn't have entries on the local member: test_map
 at com.hazelcast.sql.impl.QueryException.error(QueryException.java:68)
 at com.hazelcast.sql.impl.QueryException.error(QueryException.java:52)
 at com.hazelcast.sql.impl.QueryException.error(QueryException.java:48)
 at com.hazelcast.sql.impl.schema.map.PartitionedMapTableResolver.emptyError(PartitionedMapTableResolver.java:214)
 at com.hazelcast.sql.impl.schema.map.PartitionedMapTableResolver.createTable(PartitionedMapTableResolver.java:124)
 at com.hazelcast.sql.impl.schema.map.PartitionedMapTableResolver.getTables(PartitionedMapTableResolver.java:73)
 at com.hazelcast.sql.impl.schema.SqlCatalog.<init>(SqlCatalog.java:42)
 at com.hazelcast.sql.impl.SqlServiceImpl.prepare(SqlServiceImpl.java:290)
 ... 24 more
```

--- a/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/IMapSqlConnector.java
+++ b/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/IMapSqlConnector.java
@@ -124,7 +124,7 @@ public class IMapSqlConnector implements SqlConnector {
 
         MapService service = nodeEngine.getService(MapService.SERVICE_NAME);
         MapServiceContext context = service.getMapServiceContext();
-        MapContainer container = context.getMapContainer(externalName);
+        MapContainer container = context.getExistingMapContainer(externalName);
 
         long estimatedRowCount = estimatePartitionedMapRowCount(nodeEngine, context, externalName);
         boolean hd = container != null && container.getMapConfig().getInMemoryFormat() == InMemoryFormat.NATIVE;

--- a/hazelcast/src/main/java/com/hazelcast/map/impl/MapServiceContext.java
+++ b/hazelcast/src/main/java/com/hazelcast/map/impl/MapServiceContext.java
@@ -80,6 +80,8 @@ public interface MapServiceContext extends MapServiceContextInterceptorSupport,
 
     MapContainer getMapContainer(String mapName);
 
+    MapContainer getExistingMapContainer(String mapName);
+
     Map<String, MapContainer> getMapContainers();
 
     PartitionContainer getPartitionContainer(int partitionId);

--- a/hazelcast/src/main/java/com/hazelcast/map/impl/MapServiceContextImpl.java
+++ b/hazelcast/src/main/java/com/hazelcast/map/impl/MapServiceContextImpl.java
@@ -287,6 +287,11 @@ class MapServiceContextImpl implements MapServiceContext {
     }
 
     @Override
+    public MapContainer getExistingMapContainer(String mapName) {
+        return mapContainers.get(mapName);
+    }
+
+    @Override
     public Map<String, MapContainer> getMapContainers() {
         return mapContainers;
     }

--- a/hazelcast/src/main/java/com/hazelcast/sql/impl/schema/map/PartitionedMapTableResolver.java
+++ b/hazelcast/src/main/java/com/hazelcast/sql/impl/schema/map/PartitionedMapTableResolver.java
@@ -103,7 +103,7 @@ public class PartitionedMapTableResolver extends AbstractMapTableResolver {
         String name
     ) {
         try {
-            MapContainer mapContainer = context.getMapContainer(name);
+            MapContainer mapContainer = context.getExistingMapContainer(name);
 
             // Handle concurrent map destroy.
             if (mapContainer == null) {

